var/spack/repos/builtin/packages/mpifileutils/package.py

```diff
@@ -102,6 +106,12 @@ def cmake_args(self):
         else:
             args.append("-DENABLE_EXPERIMENTAL=OFF")
 
+        if "+daos" in self.spec:
```

Comment:
Is there a chance that another package that might become a dependency of this one could have a variant with this name?

We generally recommend using `self.spec.satisifies()` if the goal is to check that the variant applies to the current spec.  See the docs at https://spack.readthedocs.io/en/latest/packaging_guide.html#testing-spec-constraints:

"The difference between the two functions is that `satisfies()` tests whether spec constraints overlap at all, while `in` tests whether a spec or any of its dependencies satisfy the provided spec."

If `in` has the desired behavior, then I'll merge as is.